lib/wickedbase.pm

```diff
@@ -976,6 +976,9 @@ sub write_cfg {
     my $rand = random_string;
     # replace variables
     $content =~ s/\{\{(\w+)\}\}/$self->lookup($1, $args{env})/eg;
+    # make sure that dirs exists
+    my $dir_name = dir_name($filename);
+    assert_script_run('mkdir -p ' . $dir_name);
```

Comment:
Since we use the variable only once, and to avoid conflict between variable name and function, we could use this instead:

```suggestion
    assert_script_run('mkdir -p ' . dir_name($filename));
```